src/core/Linux.zig

```diff
@@ -107,7 +107,22 @@ pub fn initWindow(
                     else => "An unknown error occured while trying to connect to X11",
                 };
                 log.err("{s}\n\nFalling back to Wayland\n", .{err_msg});
-                try Wayland.initWindow(core, window_id);
+                Wayland.initWindow(core, window_id) catch |e| {
+                    log.err("Failed to connect to fallback display server, Wayland.\n", .{});
+                    var libs = std.ArrayList(u8).init(core.allocator);
+                    defer libs.deinit();
+                    if (X11.libx11 == null) {
+                        try libs.appendSlice("\t* " ++ X11.LibX11.lib_name ++ "\n");
+                    }
+                    if (X11.libxkbcommon == null) {
+                        try libs.appendSlice("\t* " ++ X11.LibXkbCommon.lib_name ++ "\n");
+                    }
+                    if (X11.libgl == null) {
+                        try libs.appendSlice("\t* " ++ X11.LibGL.lib_name ++ "\n");
+                    }
+                    log.err("The following X11 libraries were not available:\n{s}", .{libs.items});
```

Comment:
I think there's a way to write this without the ArrayList, but haven't thought about it too much. A `log` API that prints partial text instead of a full message? Or printing multiple log lines?